src/emqtt.erl

```diff
@@ -1073,6 +1141,53 @@ delete_inflight_when_full(Packet, State) ->
         false -> {next_state, connected, State1}
     end.
 
+expiry_inflight_msgs(Now, State = #state{inflight = Inflight}) ->
+    Fun = fun(PacketId, {Type, Msg, Meta}, Acc) ->
+              SentAt = maps:get(first_sent_at, Meta),
+              ExpiryTime = get_message_delivery_timeout(),
+              case Now - SentAt of
+                  Diff when Diff >= ExpiryTime ->
+                      Acc#{PacketId => {Type, Msg, Meta}};
+                  _ ->
+                      Acc
+              end
+          end,
+    ExpiredMsgs = maps:fold(Fun, Inflight, #{}),
+    case maps:size(ExpiredMsgs) of
+        0 -> State;
+        _ ->
+            ExpiredMsgs1 = lists:sort(
```

Comment:
this sort is unnecessary

Reply:
The main purpose is to call back the functions in the order in which the messages were sent.